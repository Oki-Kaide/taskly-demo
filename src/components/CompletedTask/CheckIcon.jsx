import React from 'react';

const CheckIcon = () => (
  <svg width="18px" height="14px" viewBox="0 0 18 14" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlnsXlink="http://www.w3.org/1999/xlink">
    <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
        <g id="Website-03" transform="translate(-763.000000, -362.000000)">
            <g id="Group-5" transform="translate(760.000000, 40.000000)">
                <g id="Group-6" transform="translate(0.000000, 100.000000)">
                    <polygon id="Path" points="0 217 24 217 24 241 0 241"></polygon>
                    <polygon id="Path" fill="#FF3339" fillRule="nonzero" points="9 233.17 4.83 229 3.41 230.41 9 236 21 224 19.59 222.59"></polygon>
                </g>
            </g>
        </g>
    </g>
  </svg>
);

export default CheckIcon;
