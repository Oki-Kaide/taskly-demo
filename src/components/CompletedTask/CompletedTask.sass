@import '../../styles/variables.sass'

.completed-task
  display: flex
  flex-direction: row
  align-items: center
  height: 24px
  margin-bottom: 15px

  span
    font-family: CircularStd
    font-size: 14px
    color: #000000
    letter-spacing: 0
    line-height: 21px
    margin-left: 10px