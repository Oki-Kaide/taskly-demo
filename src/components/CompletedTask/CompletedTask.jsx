import React from 'react';
import CheckIcon from './CheckIcon';
import './CompletedTask.sass'

const CompletedTask = ({text}) => {
  return (
    <div className="completed-task">
      <CheckIcon />
      <span>{text}</span>
    </div>
  )
}

export default CompletedTask;