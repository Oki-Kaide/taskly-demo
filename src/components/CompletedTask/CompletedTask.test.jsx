import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CompletedTask from './CompletedTask';
import CheckIcon from './CheckIcon';

const createTestProps = props => ({
  text: 'test',
  ...props,
});

describe('CompletedTask component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<CompletedTask {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<CompletedTask {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('CheckIcon component', () => {
  it('renders without crashing', () => {
    shallow(<CheckIcon />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<CheckIcon />);
    expect(toJson(component)).toMatchSnapshot();
  });
});