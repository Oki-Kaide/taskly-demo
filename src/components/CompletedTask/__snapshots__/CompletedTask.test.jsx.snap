// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckIcon component matches the rendered snapshot 1`] = `
<svg
  height="14px"
  version="1.1"
  viewBox="0 0 18 14"
  width="18px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Page-1"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Website-03"
      transform="translate(-763.000000, -362.000000)"
    >
      <g
        id="Group-5"
        transform="translate(760.000000, 40.000000)"
      >
        <g
          id="Group-6"
          transform="translate(0.000000, 100.000000)"
        >
          <polygon
            id="Path"
            points="0 217 24 217 24 241 0 241"
          />
          <polygon
            fill="#FF3339"
            fillRule="nonzero"
            id="Path"
            points="9 233.17 4.83 229 3.41 230.41 9 236 21 224 19.59 222.59"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`CompletedTask component matches the rendered snapshot 1`] = `
<div
  className="completed-task"
>
  <CheckIcon />
  <span>
    test
  </span>
</div>
`;
